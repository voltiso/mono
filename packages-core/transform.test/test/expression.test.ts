// â €â“¥ 2022     ðŸŒ©    ðŸŒ©     â €   â €
// â €         ðŸŒ© VÍ›oÍ›Í›Í›ltÍ›Í›Í›iÍ›Í›Í›Í›soÍ›Í›Í›.comâ €  â €â €â €

import * as fs from 'node:fs/promises'

describe('expression', () => {
	it('works', async () => {
		expect.hasAssertions()

		const file = await fs.readFile('./dist/cjs/expression.d.ts')
		expect(file.toString()).toMatchSnapshot()
	})
})
