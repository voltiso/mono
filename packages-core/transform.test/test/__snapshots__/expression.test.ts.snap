// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expression works 1`] = `
"export declare class NonGeneric {
    Type: string;
}
export interface Generic<T> {
    get Type(): T;
}
export declare type A = NonGeneric['Type'];
export declare type A_ = /* NonGeneric['Type'] */ string;
export declare type G0<T> = Generic<T>['Type'];
export declare type G0_<T> = /* Generic<T>['Type'] */ T;
export declare type G1 = Generic<string>['Type'];
export declare type G1_ = /* Generic<string>['Type'] */ string;
//# sourceMappingURL=expression.d.ts.map"
`;
